src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -37,75 +51,60 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
       const optionIndex = getDefaultIndex(defaultOption, options)
       onSave({
         text: optionsDisplayProperty
-          ? options[optionIndex][optionsDisplayProperty]
+          ? (options[optionIndex] as ObjectOption)[optionsDisplayProperty]
           : options[optionIndex],
         selection: options[optionIndex],
         optionIndex,
+        isCustomOption: false,
       })
       setSelectedIndex(optionIndex)
     }
-    if (currentResponse?.text && currentResponse.optionIndex && currentResponse.optionIndex>options.length) {
-      addItemHandler(_, currentResponse.text)
-      return
-    }
     if (currentResponse?.text) {
       const { optionIndex } = currentResponse
       setSelectedIndex(optionIndex)
     }
-    
   }, [])
 
-
-  //addItemHandler is called when new data is entered. This in turn calls original hook to re-render displayed options
-  const addItemHandler = (e: any, data: string) => {
-    const  newOptionText  = data
-    const index = dropDownOptions[0].text !== undefined? dropDownOptions.length : currentResponse.optionIndex
-    const newOption = {
-      [optionsDisplayProperty]: newOptionText,
-    }
-    const newAddedOptions : any = [...addedOptions, newOption]
-    setAddedOptions(newAddedOptions)
-    setSelectedIndex(index)
+  const addItemHandler = (text: string) => {
+    setAddedOption(text)
+    const newIndex = options.length
+    setSelectedIndex(newIndex)
+    handleChange(newIndex, text)
   }
 
-  function handleChange(_: any, data: any) {
-      var { value: optionIndex } = data
-      //If condition saves appropriate value if new value is entered by user
-        if (typeof(optionIndex) === 'string') {
-          const optionIndexForNewValues = dropDownOptions.length
-          onSave({
-            text: optionIndex,
-            selection: {
-              key: `${optionIndexForNewValues}_${optionIndex}`,
-              text: optionIndex,
-              value: optionIndexForNewValues,
-            },
-            optionIndex: optionIndexForNewValues,
-          })
-          return
-        }
-        setSelectedIndex(optionIndex === '' ? undefined : optionIndex)
-        if (optionIndex !== '') {
-          onSave({
-            text: (optionsDisplayProperty[optionIndex][optionsDisplayProperty] !== undefined && optionsDisplayProperty)
-              ? dropDownOptions[optionIndex][optionsDisplayProperty]
-              : dropDownOptions[optionIndex]['text'],
-            selection: dropDownOptions[optionIndex],
-            optionIndex,
-          })
-        }
-        // Reset response if selection cleared
-        else onSave(null)
-  }
+  const handleChange = (value: number | string, addedText?: string) => {
```

Comment:
The reason why I've added the second (optional) parameter here is because we want to run `handleChange` from `addItemHandler`. And because it runs as part of the same component render, the value of "addedOption" is not reliably available just after it's set. So we just pass it in instead.

In fact, we can almost do without having `addedOption` as a state value at all, since it can be generated from the response as part of "getDropdownOptions". The only time it's really needed is so its value can persist while another option is selected (I think).